network/src/legacy/gossip/attestation.rs

```diff
@@ -205,9 +223,16 @@ impl View {
 					}
 				};
 
+				let signing_context = match self.signing_context(&message.relay_chain_leaf) {
```

Comment:
I would instead recommend to move this logic into the `fn check_statement`, where the `MessageValidationData` will be available and the `fn signing_context` will be unnecessary